File: Kotlin/src/LongestCommonPrefix.kt
/**
 * Write a function to find the longest common prefix string amongst an array of strings.
 *
 * Accepted.
 */
class LongestCommonPrefix {

    fun longestCommonPrefix(strs: Array<String>): String {
        if (strs.isEmpty()) {
            return ""
        }
        if (strs.size == 1) {
            return strs[0]
        }

        (0 until strs[0].length).forEach { i ->
            (1 until strs.size)
                    .filter {
                        i == strs[it].length || strs[0][i] != strs[it][i]
                    }
                    .forEach {
                        return strs[0].substring(0, i)
                    }
        }

        return strs[0]
    }

}

File: Python/LongestCommonPrefix.py
# -*- coding: UTF-8 -*-
# Write a function to find the longest common prefix string amongst an array of strings.
#
# Python, Python 3 all accepted.


class LongestCommonPrefix(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        if strs is None or len(strs) == 0:
            return ""
        if len(strs) == 1:
            return strs[0]

        for i in range(0, len(strs[0])):
            for j in range(1, len(strs)):
                if i == len(strs[j]) or strs[0][i] != strs[j][i]:
                    return strs[0][0:i]

        return strs[0]
